examples/diffusion/laplace_fluid_2d.py

```diff
@@ -0,0 +1,103 @@
+r"""
+A Laplace equation that models the flow of "dry water" around an obstacle
+shaped like a Citroen CX.
+
+
+Description
+-----------
+
+As discussed e.g. in the Feynman lectures in Section 12-5 of Volume 2
+(https://www.feynmanlectures.caltech.edu/II_12.html#Ch12-S5),
+the flow of an irrotational and incompressible fluid can be modelled with a
+potential :math:`\ul{v} = \ul{grad}(\phi)` that obeys
+
+.. math::
+    \nabla \cdot \ul{\nabla}\,\phi = \Delta\,\phi = 0
+
+The weak formulation for this problem is to find :math:`\phi` such that:
+
+\int_{\Omega} \nabla \psi \cdot \nabla \phi
+    = \int_{\Gamma_{left}} \ul{v}_0 \cdot n  \, \psi
+    + \int_{\Gamma_{right}} \ul{v}_0 \cdot n  \, \psi
+    + \int_{\Gamma_{top}} \ul{v}_0 \cdot n \,\psi
+    + \int_{\Gamma_{bottom}} \ul{v}_0 \cdot n \, \psi
+    \;, \quad \forall \psi \;,
+
+where :math:`\ul{v}_0` is the 2D vector defining the far field velocity that
+generates the incompressible flow.
+
+Since the value of the potential is defined up to a constant value, a Dirichlet
+boundary condition is set on one edge to avoid having a singular problem.
+
+Usage examples
+--------------
+
+Run with simple.py script::
+
+    python3 simple.py examples/diffusion/laplace_fluid_2d.py
+    python3 resview.py citroen.vtk
+
+
+Generating the mesh
+-------------------
+
+The mesh can be generated with::
+    gmsh -2 -f msh22 meshes/2d/citroen.geo -o meshes/2d/citroen.msh
+    python script/convert_mesh.py --2d meshes/2d/citroen.msh \
+        meshes/2d/citroen.h5
+"""
+from sfepy import data_dir
+
+filename_mesh = data_dir + '/meshes/2d/citroen.h5'
+
+v0 = np.array([1, 0.25])
+
+materials = {
+    'm': ({'sigma': 1.0},),
+    'f': ({'left': np.dot(v0.T, np.array([-1, 0])),
+           'right': np.dot(v0.T, np.array([1, 0])),
+           'top': np.dot(v0.T, np.array([0, 1])),
+           'bottom': np.dot(v0.T, np.array([0, -1]))},),
+}
+
+regions = {
+    'Omega': 'all',
+    'Gamma_Left': ('vertices in (x < 0.1)', 'facet'),
+    'Gamma_Right': ('vertices in (x > 1919.9)', 'facet'),
+    'Gamma_Top': ('vertices in (y > 917.9)', 'facet'),
+    'Gamma_Bottom': ('vertices in (y < 0.1)', 'facet'),
+    'Vertex': ('vertex in r.Gamma_Left', 'vertex'),
+}
+
+fields = {
+    'u': ('real', 1, 'Omega', 1),
+}
+
+variables = {
+    'phi': ('unknown field', 'u', 0),
+    'psi': ('test field', 'u', 'phi'),
+}
+
+# these EBCS prevent the matrix from being singular
+ebcs = {
+    'fix': ('Vertex', {'phi.0': 0.0}),
+}
+
+integrals = {
+    'i': 2,
+}
+
+equations = {
+    'Laplace equation':
+        """dw_laplace.i.Omega( m.sigma, psi, phi )
+         = dw_integrate.i.Gamma_Left(f.left, psi)
+         + dw_integrate.i.Gamma_Right(f.right, psi)"""
```

Comment:
The top and bottom regions and material are not used?

Reply:
Please see my next version (I forgot to re-test after copying from my main script) where the domains are used :)